packet/checksum_test.go

```diff
@@ -39,146 +38,134 @@ CalculatePseudoHdrIPv6UDPCksum
-	pkt := packet.GetPacket()
-	packet.InitEmptyIPv4UDPPacket(pkt, payloadSize)
-	initIPv4Addrs(pkt)
+	pkt := GetPacket()
+	InitEmptyIPv4UDPPacket(pkt, payloadSize2)
+	initIPv4Addrs2(pkt)
 	initPorts(pkt)
 	initData(pkt)
 
-	ipcksum := packet.CalculateIPv4Checksum(pkt.GetIPv4())
-	pkt.GetIPv4().HdrChecksum = packet.SwapBytesUint16(ipcksum)
+	ipcksum := CalculateIPv4Checksum(pkt.GetIPv4())
+	pkt.GetIPv4().HdrChecksum = SwapBytesUint16(ipcksum)
 
 	want := wantIPv4UDP
-	got := packet.CalculateIPv4UDPChecksum(pkt.GetIPv4(), pkt.GetUDPForIPv4(), pkt.Data)
+	got := CalculateIPv4UDPChecksum(pkt.GetIPv4(), pkt.GetUDPForIPv4(), pkt.Data)
 	if got != want {
 		t.Errorf("Incorrect result:\ngot: %x, \nwant: %x\n\n", got, want)
 	}
 }
 
 func TestCalculateIPv6UDPChecksum(t *testing.T) {
-	pkt := packet.GetPacket()
-	packet.InitEmptyIPv6UDPPacket(pkt, payloadSize)
-	initIPv6Addrs(pkt)
+	pkt := GetPacket()
+	InitEmptyIPv6UDPPacket(pkt, payloadSize2)
+	initIPv6Addrs2(pkt)
 	initPorts(pkt)
 	initData(pkt)
 
 	want := wantIPv6UDP
-	got := packet.CalculateIPv6UDPChecksum(pkt.GetIPv6(), pkt.GetUDPForIPv6(), pkt.Data)
+	got := CalculateIPv6UDPChecksum(pkt.GetIPv6(), pkt.GetUDPForIPv6(), pkt.Data)
 	if got != want {
 		t.Errorf("Incorrect result:\ngot: %x, \nwant: %x\n\n", got, want)
 	}
 }
 
 func TestCalculateIPv4ICMPChecksum(t *testing.T) {
-	pkt := packet.GetPacket()
-	packet.InitEmptyIPv4ICMPPacket(pkt, payloadSize)
-	initIPv4Addrs(pkt)
+	pkt := GetPacket()
+	InitEmptyIPv4ICMPPacket(pkt, payloadSize2)
+	initIPv4Addrs2(pkt)
 	initData(pkt)
 	initICMP(pkt.GetICMPForIPv4())
 
-	ipcksum := packet.CalculateIPv4Checksum(pkt.GetIPv4())
-	pkt.GetIPv4().HdrChecksum = packet.SwapBytesUint16(ipcksum)
+	ipcksum := CalculateIPv4Checksum(pkt.GetIPv4())
+	pkt.GetIPv4().HdrChecksum = SwapBytesUint16(ipcksum)
 
 	want := wantIPv4ICMP
-	got := packet.CalculateIPv4ICMPChecksum(pkt.GetIPv4(), pkt.GetICMPForIPv4(), pkt.Data)
+	got := CalculateIPv4ICMPChecksum(pkt.GetIPv4(), pkt.GetICMPForIPv4(), pkt.Data)
 
 	if got != want {
 		t.Errorf("Incorrect result:\ngot: %x, \nwant: %x\n\n", got, want)
 	}
 }
 
 func TestCalculateIPv6ICMPChecksum(t *testing.T) {
-	pkt := packet.GetPacket()
-	packet.InitEmptyIPv6ICMPPacket(pkt, payloadSize)
-	initIPv6Addrs(pkt)
+	pkt := GetPacket()
+	InitEmptyIPv6ICMPPacket(pkt, payloadSize2)
+	initIPv6Addrs2(pkt)
 	initData(pkt)
 	initICMP(pkt.GetICMPForIPv6())
 
 	want := wantIPv6ICMP
-	got := packet.CalculateIPv6ICMPChecksum(pkt.GetIPv6(), pkt.GetICMPForIPv6(), pkt.Data)
+	got := CalculateIPv6ICMPChecksum(pkt.GetIPv6(), pkt.GetICMPForIPv6(), pkt.Data)
 
 	if got != want {
 		t.Errorf("Incorrect result:\ngot: %x, \nwant: %x\n\n", got, want)
 	}
 }
 
-func initEtherAddrs(pkt *packet.Packet) {
-	pkt.Ether.SAddr = [common.EtherAddrLen]byte{0x11, 0x22, 0x33, 0x44, 0x55, 0x66}
-	pkt.Ether.DAddr = [common.EtherAddrLen]byte{0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f}
-}
-
-func initIPv4Addrs(pkt *packet.Packet) {
+func initIPv4Addrs2(pkt *Packet) {
```

Comment:
such amount of "2" functions and variables is abnormal
maybe  common function with parameters is needed?

Reply:
> maybe common function with parameters is needed?

exported, placed in utils_for_test.go.

Reply:
That functions is needed only in acl_test. I don't know why they are in utils really, however It is strange idea to make functions that only initialize values with parameters. I will place "l" instead of "2"